crates/ironrdp-pdu/src/rdp/suppress_output.rs

```diff
@@ -0,0 +1,57 @@
+use std::io;
+
+use byteorder::{ReadBytesExt as _, WriteBytesExt as _};
+use num_derive::{FromPrimitive, ToPrimitive};
+use num_traits::{FromPrimitive, ToPrimitive};
+
+use crate::{geometry::InclusiveRectangle, PduParsing};
+
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, FromPrimitive, ToPrimitive, Clone)]
+pub enum AllowDisplayUpdatesType {
+    SuppressDisplayUpdates = 0x00,
+    AllowDisplayUpdates = 0x01,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SuppressOutputPdu {
+    allow_display_updates: AllowDisplayUpdatesType,
+    desktop_rect: Option<InclusiveRectangle>,
+}
+
+impl PduParsing for SuppressOutputPdu {
+    type Error = io::Error;
+
+    fn from_buffer(mut stream: impl io::Read) -> Result<Self, Self::Error> {
+        let allow_display_updates = AllowDisplayUpdatesType::from_u8(stream.read_u8()?)
+            .ok_or_else(|| io::Error::new(io::ErrorKind::InvalidData, "Invalid display update type"))?;
```

Comment:
Nitpick: [here](https://github.com/Devolutions/IronRDP/blob/d69ff6e6e7ebf76864f4d812a33677b8d0d70eac/STYLE.md#formatting-of-error-messages) is the error message style we are committed to

Reply:
ok